com.unity.perception/Documentation~/Tutorial/Phase2.md

```diff
@@ -0,0 +1,222 @@
+{
+    public float minIntensity;
+    public float maxIntensity;
+
+    public void SetIntensity(float rawIntensity)
+    {
+        var light = gameObject.GetComponent<Light>();
+        if (light)
+        {
+            var scaledIntensity = rawIntensity * (maxIntensity - minIntensity) + minIntensity;
+            light.intensity = scaledIntensity;
+        }
+    }
+}
+```
+In the above code, we have created a new `SetIntensity` function that first scales the incoming intensity (assumed to be between 0 and 1) to our desired range and then assigns it to the light's intensity. The `Light` component is now fetched from the GameObject that this Randomizer tag is attached to. This works because both this tag component and the `Light` component are attached to the same object in the Scene (which is one of the lights). 
+
+This component is already added to both our lights. We now need to set our desired minimum and maximum intensities, and this can be done through the _**Inspector**_ view.
+
+* **Action**: Select `Directional Light` and from the _**Inspector** UI for the `MyLightRandomizerTag` component, set `Min Intensity` to 0.5 and `Max Intensity` to 3.
+* **Action**: Repeat the above step for `Directional Light (1)` and set `Min Intensity` to 0 and `Max Intensity` to 0.4.
+
+Note that with this change, we fully transfer the responsibility for the light's intensity range to `MyLightRandomizerTag.cs` and assume the intensity value coming from `My Light Randomizer` is always between 0 and 1. Therefore, we now need to change the range for the corresponding Parameter in `My Light Randomizer` to (0,1). 
+
+* **Action**: Select `SimulationScenario` and from the UI snippet for `My Light Randomizer`, change the range for `Light Intensity Parameter` from (0.5,3.5) to (0,1).
+
+We also need to make a minor change to `MyLightRandomizer.cs` in order to make it compatible with this new approach.
+
+* **Action**: Open `MyLightRandomizer.cs` and modify it as seen below:
+
+```
+using System;
+using UnityEngine;
+using UnityEngine.Experimental.Perception.Randomization.Parameters;
+using UnityEngine.Experimental.Perception.Randomization.Randomizers;
+
+[Serializable]
+[AddRandomizerMenu("Perception/My Light Randomizer")]
+public class MyLightRandomizer : Randomizer
+{
+    public FloatParameter lightIntensityParameter;
+    public ColorRgbParameter lightColorParameter;
+
+    protected override void OnIterationStart()
+    {
+        var taggedObjects = tagManager.Query<MyLightRandomizerTag>();
+        foreach (var taggedObject in taggedObjects)
+        {
+            var light = taggedObject.GetComponent<Light>();
+            if (light)
+            {                
+                light.color = lightColorParameter.Sample();
+            }
+
+            var tag = taggedObject.GetComponent<MyLightRandomizerTag>();
+            if (tag)
+            {
+                tag.SetIntensity(lightIntensityParameter.Sample());
+            }
+        }
```

Comment:
Two things here:
1. We can add a [RequireComponent] attribute to the MyLightRandomizerTag to ensure that a light component is attached to the object with the tag
2. We can safely remove both null checks and simplify this code a bit. The light component will be required, so it will be guaranteed to be present on the object. The MyLightRandomizerTag is certainly not null because the taggedObject was explicitly queried for this tag.